### PR TITLE
records: list of files and preview

### LINKED ISSUE #173: Landing page template should include list of files and preview
The landing page of a record (e.g. ``/records/1``) should include the list of associated files, as well as a preview of the file if possible.

Model it over Zenodo:

Screenshot 2019-08-01 at 11 31 25

We might need to configure the IIIF previewer:
https://invenio-iiif.readthedocs.io/en/latest/usage.html#preview

IMPORTANT: Read the comments below

### PR DESCRIPTION
**Pull request by mvidalgarcia**
Closes #173

Currently depends on https://github.com/inveniosoftware/invenio-records-files/compare/master...mvidalgarcia:add-files to work. Waiting for proper `_files` integration.

When displaying PDFs, some warnings/errors show up:
image

### CODE REVIEW

**Comment by mvidalgarcia** [comment]
Some more context about the PDF warnings:
- PDF with a simple `foo bar` in Arial font: same errors.
- Blank PDF: only `pdf.worker.min.js` error.

Conclusion: `data:font` error might be related to the PDF font.

File: {{cookiecutter.project_shortname}}/{{cookiecutter.package_name}}/records/templates/records/record.html [hunk 1]
```diff
@@ -29,14 +29,99 @@
...
+        <thead>
+          <tr class="">
+            <th>{{_('Name')}}</th>
+            <th>{{_('Size')}}</th>
+          </tr>
+        </thead>
+        <tbody>
+        {%- for file in files|sort(attribute='key') -%}
+        {%- set file_url_download = url_for('invenio_records_ui.recid_files', pid_value=pid.pid_value, filename=file.key, download=1) %}
+        {%- set file_url_preview = url_for('invenio_records_ui.recid_previewer', pid_value=pid.pid_value, filename=file.key) %}
```
Comment (equadon) [inline {{cookiecutter.project_shortname}}/{{cookiecutter.package_name}}/records/templates/records/record.html]:
```suggestion
        {%- set file_url_preview = url_for('invenio_records_ui.recid_preview', pid_value=pid.pid_value, filename=file.key) %}
```

File: {{cookiecutter.project_shortname}}/{{cookiecutter.package_name}}/records/templates/records/record.html [hunk 2]
```diff
@@ -29,14 +29,99 @@
...
+    <div class="panel" id="preview">
+      <div class="panel-heading">
+        <a class="panel-toggle" data-toggle="collapse" href="#collapseOne">
+          {{_('Preview')}}
+          <span class="pull-right show-on-collapsed"><i class="fa fa-chevron-right"></i></span>
+          <span class="pull-right hide-on-collapsed"><i class="fa fa-chevron-down"></i></span>
+        </a>
+      </div>
+      <div id="collapseOne" class="collapse in">
+        {{- preview_file('invenio_records_ui.recid_previewer', pid=pid, filename=selected_file.key) }}
```
Comment (equadon) [inline {{cookiecutter.project_shortname}}/{{cookiecutter.package_name}}/records/templates/records/record.html]:
```suggestion
        {{- preview_file('invenio_records_ui.recid_preview', pid=pid, filename=selected_file.key) }}
```

**Comment by roksys** [comment]
Looks good to me. Just a few questions.

1. Is it expected to have 1st file's previewer expanded by default?

Screenshot 2019-11-22 at 13 31 48

2. Have you tested all previewers? TXT previewer isn't working for me.
Screenshot 2019-11-22 at 13 32 44

**Comment by kprzerwa** [comment]
the previewer console errors should be fixed at this point (by configuration of the content-security-policy)

**Pull request merged**